src/utils/ValidationUtils.ts

```
@@ -0,0 +1,297 @@
+    }
+
+    if (database.templates && typeof database.templates !== 'object') {
+      errors.push('Database templates must be an object');
+    }
+
+    return { isValid: errors.length === 0, errors };
+  }
+
+  // Input sanitization
+  static sanitizeString(input: string): string {
+    return input.trim().replace(/[\x00-\x1F\x7F-\x9F]/g, '');
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)